web/lib/queries/common/heroFields.ts

```diff
@@ -4,7 +4,10 @@ export const heroFields = /* groq */ `{
     "type": coalesce(heroType, 'default'),
     "ratio": heroRatio,
     "isBigTitle":isBigTitle,
-    "title": select(isBigTitle => select( heroType == 'fiftyFifty' => heroBigTitleFiftyFifty, heroBigTitleDefault) , heroTitle),
+    "title": select( 
```

Comment:
i believe it is possible to rewrite this like:

```
{
  _type == 'type1' => {
    // Your selection of fields for type1
  },
  _type == 'type2' => {
    // Your selection of fields for type2
    "url": file.asset->url // Use joins to get data of referenced document
  }
}
```
instead of using nested selects